### PR TITLE
#162163229 Users should be able to login via Facebook, Google

### PR DESCRIPTION
**Pull request by bibangamba**
#### What does this PR do?
Enable Author's Haven users login into the application using Google or Facebook.

#### Description of tasks to be completed?
- Install Google and Facebook packages to get tokens from the corresponding platform
- Make a call to the backend API endpoint to validate/process the retrieved `access_token`
- Log the user in (on success), save returned JWT, and redirect the user to the landing (articles) page OR show error messages if the login failed.
 
#### How can this be manually tested?
Open the PR review app and click on a social login button. On the first try, this should open a pop-up browser where the social platform(s) asks you to authorize our app by logging it clicking authorize.

#### What are the relevant pivotal tracker stories?
[#162163229](https://www.pivotaltracker.com/story/show/162163229)

#### Screenshots
Facebook pop-up
image

Successful Facebook login
image

Google pop-up
image

Successful Google login
image

### CODE REVIEW

File: src/App.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { shallow } from "enzyme";
+import App from "./App";
+
+it("renders without crashing", () => {
+  // const div = document.createElement("div");
+  const app = shallow(<App />);
+  expect(app).toMatchSnapshot();
+  // ReactDOM.render(shallow(<App />), div);
```
Comment (JaymesKat) [inline src/App.spec.jsx]: Remove commented code if no longer needed @bibangamba

**Review by JaymesKat — CHANGES_REQUESTED** [review]

**Comment by Kasulejoseph** [comment]
great work andrew. But for a success login a user should be redirected to the `home page` not `/articles` page. you should edit the commit message also.

**Review by Kasulejoseph — CHANGES_REQUESTED** [review]
I think something is wrong with your buttons (google and facebook) for both login and signup, they dont respond to click actions. please fix it .

**Comment by bibangamba** [comment]
Thank you @Kasulejoseph, found the problem. I'm using environment variable on my buttons and I forgot to set that up in the app.json and add the variables inside the Heroku config variables

**Review by Kasulejoseph — APPROVED** [review]

File: app.json [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
+  "buildpacks": [{
       "url": "heroku/nodejs"
     },
     {
       "url": "https://github.com/mars/create-react-app-buildpack"
     }
   ],
   "stack": "heroku-18"
-}
+}
```
Comment (JaymesKat) [inline app.json]: add new lines at the end of files.

Comment (bibangamba) [reply app.json]: I had a problem with the extension that was helping me do this (blank line). it excluded `.json` files and other types. Enabling it on these files didn't help. I solved it by changing vs code settings to add a new line at the end of a file.

**Review by KicaRonaldOkello — APPROVED** [review]
I think it now looks good

**Review by joshtrigger — APPROVED** [review]

**Review by SokoPaulSokool — APPROVED** [review]

**Review by kafuuma — APPROVED** [review]
Nice work Andrew
I have tested and both facebook and google works well
This is good work

**Review by JaymesKat — APPROVED** [review]
LGTM

**Pull request merged**